dora/core/client/fs/src/main/java/alluxio/client/file/PositionReadFileInStream.java

```diff
@@ -109,6 +119,13 @@ private int fillWithCache(long targetStartPos, ByteBuffer outBuffer) {
      * @return number of bytes that's been prefetched, 0 if exception occurs
      */
     private int prefetch(PositionReader reader, long pos, int minBytesToRead) {
+      if (mDataPreloadEnabled && mURIStatus.getLength() > mDataPreloadFileSizeThreshold
+          && mURIStatus.getInAlluxioPercentage() != 100) {
+        mClient.cacheData(
+            mURIStatus.getUfsPath(), pos,
```

Comment:
record the last end position that a cache request has been sent for, so that no requests will be sent for overlapping ranges.

Reply:
this stream supports seeking backwards. Will this cause duplicated cache requests be sent?

Reply:
this is a fire & forget job and worker does the dedup